File: Models/Stics_soil_temperature/src/java/Stics_soil_temperature/SoiltempExogenous.java
import  java.io.*;
import  java.util.*;
import java.time.LocalDateTime;

public class SoiltempExogenous
{
    private Double max_temp;
    private Double min_temp;
    private Double prev_canopy_temp;
    private Double min_air_temp;
    
    public SoiltempExogenous() { }
    
    public SoiltempExogenous(SoiltempExogenous toCopy, boolean copyAll) // copy constructor 
    {
        if (copyAll)
        {
            this.max_temp = toCopy.getmax_temp();
            this.min_temp = toCopy.getmin_temp();
            this.prev_canopy_temp = toCopy.getprev_canopy_temp();
            this.min_air_temp = toCopy.getmin_air_temp();
        }
    }
    public Double getmax_temp()
    { return max_temp; }

    public void setmax_temp(Double _max_temp)
    { this.max_temp= _max_temp; } 
    
    public Double getmin_temp()
    { return min_temp; }

    public void setmin_temp(Double _min_temp)
    { this.min_temp= _min_temp; } 
    
    public Double getprev_canopy_temp()
    { return prev_canopy_temp; }

    public void setprev_canopy_temp(Double _prev_canopy_temp)
    { this.prev_canopy_temp= _prev_canopy_temp; } 
    
    public Double getmin_air_temp()
    { return min_air_temp; }

    public void setmin_air_temp(Double _min_air_temp)
    { this.min_air_temp= _min_air_temp; } 
    
}

File: Models/Stics_soil_temperature/src/cs/Stics_soil_temperature/SoiltempExogenous.cs
using System;
using System.Collections.Generic;

public class SoiltempExogenous 
{
    private double _max_temp;
    private double _min_temp;
    private double _prev_canopy_temp;
    private double _min_air_temp;
    
        public SoiltempExogenous() { }
    
    
    public SoiltempExogenous(SoiltempExogenous toCopy, bool copyAll) // copy constructor 
    {
    if (copyAll)
    {
    
    _max_temp = toCopy._max_temp;
    _min_temp = toCopy._min_temp;
    _prev_canopy_temp = toCopy._prev_canopy_temp;
    _min_air_temp = toCopy._min_air_temp;
    }
    }
    public double max_temp
        {
            get { return this._max_temp; }
            set { this._max_temp= value; } 
        }
    public double min_temp
        {
            get { return this._min_temp; }
            set { this._min_temp= value; } 
        }
    public double prev_canopy_temp
        {
            get { return this._prev_canopy_temp; }
            set { this._prev_canopy_temp= value; } 
        }
    public double min_air_temp
        {
            get { return this._min_air_temp; }
            set { this._min_air_temp= value; } 
        }
}